packages/tailwindcss/src/index.ts

```diff
@@ -28,6 +33,33 @@ export function compile(css: string): {
   walk(ast, (node, { replaceWith }) => {
     if (node.kind !== 'rule') return
 
+    // Track all user-defined classes for `@apply` support
+    if (
+      containsAtApply &&
+      // Verify that it is a valid applyable-class. An applyable class is a
+      // class that is a very simple selector, like `.foo` or `.bar`, but doesn't
+      // contain any spaces, combinators, pseudo-selectors, pseudo-elements, or
+      // attribute selectors.
+      node.selector[0] === '.' &&
```

Comment:
I believe you don't need `node.selector[0] === '.'` ?

Because inside `isSimpleClassSelector`, you perform this test with an early return:
`if (selector[0] !== '.') return false`

I would suggest to remove this line
```suggestion
```

Reply:
This is still required, if you comment this out a test should fail. This is basically for the scenario where you have `.foo.bar` which contains 2 classes which is not allowed.

Reply:
Good catch 😉